test/test_dynamic_shapes.py

```diff
@@ -678,6 +681,51 @@ def test_statically_known_true(self):
         # No guards should be generated
         self.assertEqual(len(shape_env.guards), 0)
 
+    def test_ephemeral_source_simplification(self):
+        from torch._dynamo.source import EphemeralSource
+
+        for x_first_in_check in [False, True]:
+            shape_env = ShapeEnv()
+            shape = (5, 10)
+            dynamic_dims = [DimDynamic.DYNAMIC for _ in shape]
+            x = create_symbolic_tensor(
+                "x",
+                torch.randn(*shape),
+                shape_env,
+                source=EphemeralSource(),
+                dynamic_dims=dynamic_dims,
+            )
+            y = create_symbolic_tensor(
+                "y",
+                torch.randn(*shape),
+                shape_env,
+                dynamic_dims=dynamic_dims,
+            )
+
+            def _get_ephemeral_source_symbols(t):
+                return [
+                    s.node._expr for s in (*t.shape, *t.stride(), t.storage_offset())
+                    if isinstance(s, torch.SymInt) and s.node._expr in shape_env.var_to_sources
```

Comment:
are you sure about direct access to `_expr` here? I recently had to fix a bunch of code where they were accessing `_expr` and it was wrong, they should have accessed `expr`.

Reply:
yeah I should definitely use `expr`. using that exhibits the behavior I originally expected where post-check none of the symints on x's metadata have ephemeral sources